scrapy/core/scheduler.py

```diff
@@ -99,28 +124,28 @@ def _dqpush(self, request):
             return True
 
     def _mqpush(self, request):
-        self.mqs.push(request, -request.priority)
+        self.mqs.push(request, self.request_key(request))
 
     def _dqpop(self):
         if self.dqs:
             d = self.dqs.pop()
             if d:
                 return request_from_dict(d, self.spider)
 
-    def _newmq(self, priority):
+    def _newmq(self, key):
         return self.mqclass()
 
-    def _newdq(self, priority):
-        return self.dqclass(join(self.dqdir, 'p%s' % priority))
+    def _newdq(self, key):
+        return self.dqclass(join(self.dqdir, _make_file_safe('k%s' % key)))
```

Comment:
hm, actually this is backwards incompatible - after this PR scrapy won't be able to resume crawls from earlier Scrapy versions

Reply:
I put this into the base class.